### PR TITLE
fix typos (using codespell)

### PR DESCRIPTION
**Pull request by mathause**
```bash
codespell --skip=".git,devel,_build" --ignore-words-list "varn, THIRDPARTY" . --write-changes -i 3
```

### CODE REVIEW

File: mesmer/mesmer_x/OLD_train_l_distrib.py [hunk 1]
```diff
@@ -741,9 +741,11 @@ def __init__(
             self.boundaries_params = boundaries_params
         self.boundaries_coeffs = boundaries_coeffs  # this one is more technical
         self.xtol_req = xtol_req
-        self.maxiter = maxiter  # used to have np.inf, but sometimes, the fit doesnt work... and it is meant to.
+        # used to have np.inf, but sometimes, the fit does not work... and it is meant to.
```
Comment (veni-vidi-vici-dormivi) [inline mesmer/mesmer_x/OLD_train_l_distrib.py:744]: This you did manually right, not codespell?

File: mesmer/mesmer_x/train_utils_mesmerx.py [hunk 2]
```diff
@@ -287,7 +287,7 @@ def correct_expr_parameters(self):
                         else:
                             raise ValueError(
                                 f"The term {t} appears in {self.parameters_expressions[param]},"
-                                " but couldnt find an equivalent in numpy or math."
+                                " but couldn't find an equivalent in numpy or math."
```
Comment (veni-vidi-vici-dormivi) [inline mesmer/mesmer_x/train_utils_mesmerx.py:290]: Interesting that it changes doesnt to does not but couldnt to couldn't. 😄

**Review by veni-vidi-vici-dormivi — APPROVED** [review]

**Comment by veni-vidi-vici-dormivi** [comment]
That's really cool thanks!

File: mesmer/mesmer_x/OLD_train_l_distrib.py (hunk 1 above)
Comment (mathause) [reply mesmer/mesmer_x/OLD_train_l_distrib.py:744]: yes

File: mesmer/mesmer_x/train_utils_mesmerx.py (hunk 2 above)
Comment (mathause) [reply mesmer/mesmer_x/train_utils_mesmerx.py:290]: True - it offered a selection for doesent but not for couldnt :man_shrugging:

**Pull request merged**